File: semester 4/MPP/JFTS/FestivalService/src/main/java/ro/mpp/authenticator/Authenticator.java
package ro.mpp.authenticator;

import org.springframework.stereotype.Service;
import ro.mpp.domain.User;
import ro.mpp.repository.IUserRepository;

import java.util.Optional;

@Service
public class Authenticator implements IAuthenticator {
    private final IUserRepository userRepo;

    public Authenticator(IUserRepository userRepo) {
        this.userRepo = userRepo;
    }

    public Optional<User> authenticate(String username, String password) {
        return userRepo.findByUsernameAndPassword(username, password);
    }
}

File: semester 4/MPP/C#FTS/Model/Service/Authenticator/Authenticator.cs
using C_FTS.Domain;
using C_FTS.Repository;
using C_FTS.Repository.DBRepository;
using System;
using System.Collections.Generic;
using System.Text;

namespace C_FTS.Service.Authenticator
{
    public class Authenticator : IAuthenticator
    {
        private readonly IUserRepository _userRepository = new UserRepository();
        public User? Authenticate(string username, string password)
        {
            return _userRepository.FindByUsernameAndPassword(username, password);
        }
    }
}
